src/AppShed/Extensions/StorageBundle/Controller/ApiController.php

```diff
@@ -0,0 +1,444 @@
+        $appParams = $this->getExtensionAppParameters($request);
+        if ($api->getApp() != $app) {
+            throw new NotFoundHttpException("Api with uuid $uuid not found");
+        }
+
+        $em = $this->getDoctrine()->getManager();
+        $em->remove($api);
+        $em->flush();
+
+        return $this->redirect($this->generateUrl('api_list', $appParams));
+    }
+
+    /**
+     * @Route("/api/{uuid}", name="api_show")
+     * @param Api $api
+     * @ParamConverter("api", class="AppShed\Extensions\StorageBundle\Entity\Api", options={"uuid"="uuid"})
+     * @Method({"GET", "POST"})
+     * @return JsonResponse
+
+     */
+    public function showAction(Request $request, Api $api)
+    {
+        $result = [];
+        switch ($api->getAction()) {
+            case Api::ACTION_SELECT: {
+                $result = $this->selectData($api, $request);
+            } break;
+            case Api::ACTION_INSERT: {
+                $result = $this->insertData($api, $request);
+            } break;
+            case Api::ACTION_UPDATE: {
+                $result = $this->updateData($api, $request);
+            } break;
+            case Api::ACTION_DELETE: {
+                $result = $this->deleteData($api, $request);
+            } break;
+        }
+        return new JsonResponse($result, 200);
+    }
+
+    private function deleteData(Api $api, Request $request) {
+        $result = [];
+
+        $filters = $request->request->get('filters', '');
+        $additionalFilters = $this->getAdditionalFilters($filters);
+        //GET FILTERED DATA
+        $storeData = $this->getDoctrine()->getManager()->getRepository('AppShedExtensionsStorageBundle:Data')->getDataForApi($api, $additionalFilters);
+
+        if ($api->getLimit()) {
+            $storeData = $this->limitResults($storeData, $api->getLimit());
+        }
+
+        $executed = 0;
+        if (count($storeData)) {
+            $em = $this->getDoctrine()->getManager();
+            foreach ($storeData as $k => $value) {
+                $em->remove($storeData[$k]);
+            }
+            $executed = 1;
+            $em->flush();
+        }
+        return ['value' => $executed];
+    }
+
+
+    private function updateData(Api $api, Request $request) {
+        $result = [];
+        //get this data here (before getDataForApi()) because $em->clear() make bad
+        $storeColumns = $api->getStore()->getColumns();
+        $appId = $api->getApp()->getAppId();
+
+
+        $filters = $request->request->get('filters', '');
+        $additionalFilters = $this->getAdditionalFilters($filters);
+        //GET FILTERED DATA
+        $storeData = $this->getDoctrine()->getManager()->getRepository('AppShedExtensionsStorageBundle:Data')->getDataForApi($api, $additionalFilters);
+
+        if ($api->getLimit()) {
+            $storeData = $this->limitResults($storeData, $api->getLimit());
+        }
+        $updateData = $request->request->all();
```

Comment:
Look at how the origonal code gets the data, using the AppShedApi functions - this is to filter out the `appid`, `appsecret`...